### DISCUSSION TITLE
Is tree-sitter the best tool for non-incremental parsing

### DISCUSSION THREAD

**Discussion by patrick-nicodemus** [tree-sitter/tree-sitter#3429] [Q&A]
I understand that tree-sitter's strength is in incremental parsing. Is tree-sitter still a good tool for the job if we are interested in non-incremental parsing? The documentation says that tree-sitter's parser for Rust is three times slower than that of rustc. This is a useful point of comparison, but that is a handwritten parser and that is a lot of time and energy. Are there benchmarks on tree-sitter's performance in non-incremental parsing compared to other parser-generators such as Bison, Antlr or Yacc?

**Comment by amaanq** [comment]
Not that I know of, but it'd be really cool to have such comparisons.
